content/blog/fish-extract.md

```diff
@@ -0,0 +1,116 @@
++++
+title = "`extract` in fish"
+description = ""
+tags = ["development", "fish", "shell"]
+categories = ["Development"]
+# series = []
+date = 2019-11-14T10:40:55-05:00
+draft = false
++++
+I recently switched from `zsh` to `fish` as my shell of choice.
+I liked the idea of starting from scratch, with the sane defaults that `fish` provides, as my [`zsh` configuration files were getting a bit out of control](https://github.com/svanburen/dotfiles/blob/master/zshrc).
+
+For the most part, the transition was fairly painless and straightforward.
+However, I still missed a few of the niceties that I had been used to from `zsh` over the years - one of which was the [`extract` function provided in oh-my-zsh](https://github.com/robbyrussell/oh-my-zsh/blob/master/plugins/extract/extract.plugin.zsh).
+Even after moving away from oh-my-zsh as my plugin manager in `zsh`, I had used my new plugin manager to grab [just that plugin](https://github.com/svanburen/dotfiles/blob/74dd7a02b83ca1874d721e242e0f466ca1f65692/zshrc#L13-L14) for my usage.
+
+After a cursory google, it seemed like a `fish` port of the plugin didn't exist, so I decided to try to port over the plugin myself.
+You can find the whole function [here](https://github.com/svanburen/dotfiles/blob/9e62163c674f3fef58a12d752daa78b4c5eeecbe/config.fish#L65-L125).
+
+First, we'll define a function named `extract` and give it a description.
+I've also noted in a comment where this function was ported from.
+```fish
+function extract -d "extract files from archives"
+    # largely adapted from https://github.com/robbyrussell/oh-my-zsh/blob/master/plugins/extract/extract.plugin.zsh
+```
+
+The function first checks if we have arguments - if we haven't, there's nothing to do, so we'll echo a usage string to stderr and exit.
+```fish
+    # no arguments, write usage
+    if test (count $argv) -eq 0
+        echo "Usage: extract [-option] [file ...]\n Options:\n -r, --remove    Remove archive after unpacking." >&2
+        exit 1
+    end
+```
+
+Next, we handle if we've been passed a `-r` or `--remove` option as the first argument.
+If we have, we'll remove the archive files after they've been successfully unarchived.
```

Comment:
SHOULD THIS SAY "IF WE DO"?

Reply:
I AM WONDERING IF YOU SHOULD CONTINUE ON WITH THIS "WE" BUSINESS.  PERHAPS YOU WANT TO BULLET POINT OR LIST OUT THE STEPS AS INSTRUCTIONS?  IT WOULD READ LESS REPETITIVELY.

Reply:
yeah, I'm a little tempted to move away from the "we" business, hmm